command/certificate/needsRenewal.go

```
@@ -0,0 +1,135 @@
+		Action:    cli.ActionFunc(needsRenewalAction),
+		Usage:     `Check if a certificate needs to be renewed`,
+		UsageText: `**step certificate needs-renewal** <crt_file> or <host_name> [**--expires-in <duration>]`,
+		Description: `**step certificate needs-renewal** Checks certificate expiration from file or from a host if 
+		the certificate is over 66% of its lifetime. If the certificate needs renewal this command will return '0'.
+		If the certificate is not far enough into its life, it will return '1'. If validation fails, or if an error occurs, 
+		this command will produce a non-zero return value.
+## POSITIONAL ARGUMENTS
+
+<crt_file>
+: The path to a certificate to validate.
+
+<host_name>
+: Address of remote host 
+
+## EXIT CODES
+
+This command returns 0 if needing renewal, or returns 1 if the certificate does not need renewal. It will return 255 if any errors occurred
+
+## EXAMPLES
+Check certificate for renewal using custom directory 
+'''
+$ step certificate needs-renewal ./certificate.crt 
+'''
+Check if certificate will expire within a given time 
+'''
+$ step certificate needs-renewal ./certificate.crt --expires-in 1h15m
+'''
+`,
+		Flags: []cli.Flag{
+			cli.StringFlag{
+				Name:  "expires-in",
+				Usage: `Check if the certificate expires in given time duration`,
+			},
+		},
+	}
+}
+
+func needsRenewalAction(ctx *cli.Context) error {
+	if err := errs.NumberOfArguments(ctx, 1); err != nil {
+		return err
+	}
+	var (
+		crtFile    = ctx.Args().Get(0)
+		expiresIn  = ctx.String("expires-in")
+		roots      = ctx.String("roots")
+		serverName = ctx.String("servername")
+		cert       *x509.Certificate
+	)
+
+	if addr, isURL, err := trimURL(crtFile); err != nil {
+		return err
+	} else if isURL {
+		peerCertificates, err := getPeerCertificates(addr, serverName, roots, false)
+		if err != nil {
+			os.Exit(255)
+		}
+		cert = peerCertificates[0]
+
+	} else {
+		crtBytes, err := ioutil.ReadFile(crtFile)
+		if err != nil {
+			os.Exit(255)
```

We should return an error here too

---

We should return wrapped errors here and in all the exit conditions. For example this should be like:
```
crtBytes, err := ioutil.ReadFile(crtFile)
if err != nil {
    return errors.Wrapf(err, "error reading %s", crtFile)
}
```
Or use the helper `errs.FileError(err, crtFile)`